core/src/saros/account/XMPPAccountStore.java

```diff
@@ -355,32 +350,30 @@ public void setAccountActive(XMPPAccount account) {
   }
 
   /**
-   * Deletes an account.
+   * Deletes an account from the store. If this was the default account the default account is set
+   * to <code>null</code>.
    *
    * @param account the account to delete
    * @throws IllegalArgumentException if the account is not found in the store
-   * @throws IllegalStateException if the account is active
    */
   public void deleteAccount(XMPPAccount account) {
     synchronized (this) {
-      if (!accounts.contains(account))
+      if (!accounts.remove(account))
         throw new IllegalArgumentException(
             "account '" + account + "' is not in the current account store");
 
-      if (this.activeAccount == account)
-        throw new IllegalStateException(
-            "account '" + account + "' is active and cannot be deleted");
-
-      accounts.remove(account);
+      if (defaultAccount == account) defaultAccount = null;
```

Comment:
Is there a specific reason to compare the object references instead of using `equals()`? (See [Codacy issues](https://app.codacy.com/app/Saros/saros/pullRequest?prid=3938137&bid=13677390)) Same goes for l.417.
(If its just to avoid the nullcheck, you can use `Objects.equals(Object, Object)` instead.)

Reply:
Because in this scope using == or equals will result in the same behavior. This will of course break if you make the XMPPAccount CTOR public.

Reply:
I would still prefer to use `equals(Object)` or `Objects.equals(Object, Object)` to be on the safe side. The performance loss should not be of concern.